### PR TITLE
feat(new MCP server): add Code Documentation Generation MCP Server

### PR DESCRIPTION
**Pull request by jimini55**
Fixes

### Changes

Added the Code Documentation Generation MCP Server that automatically analyzes repository structure and generates comprehensive documentation for code projects. This server leverages repomix to extract project structure and creates tailored documentation based on project type.

The server provides four key tools:
- `prepare_repository`: Analyzes project directories and extracts structure
- `create_context`: Creates documentation context from project analysis
- `plan_documentation`: Creates documentation plan based on project analysis
- `generate_documentation`: Generates document templates based on the plan

### User experience

Before this change:
- Users had to manually analyze repository structure to understand code organization
- Documentation creation was a manual process requiring significant effort
- No standardized approach to creating comprehensive project documentation
- Inconsistent documentation structure across projects

After this change:
- Users can automatically analyze repository structure with the `prepare_repository` tool
- Standardized approach to creating comprehensive documentation based on project type
- Documentation follows best practices with appropriate sections for README, API docs, etc.
- Integration with other MCP servers (like AWS Diagram MCP Server) for enhanced documentation
- Consistent documentation structure across projects
- Renamed code-doc-generation-mcp-server to code-doc-gen-mcp-server. Tested with the new name (I will use Code Documentation Generation MCP Server for README, but in code/directory code-doc-gen)

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] I have reviewed the [contributing guidelines](https://github.com/awslabs/mcp/blob/main/CONTRIBUTING.md)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented

Is this a breaking change? (N)

**RFC issue number**: N/A

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/mcp/blob/main/LICENSE).

### CODE REVIEW

File: src/code-doc-generation-mcp-server/awslabs/code_doc_generation_mcp_server/server.py [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+from pydantic import BaseModel, Field
+from typing import List, Optional
+
+
+# Configure loguru with more detailed debug output
+logger.remove()  # Remove default handler
+logger.configure(
+    handlers=[
+        {'sink': sys.stderr, 'level': 'DEBUG'},
+        {'sink': 'code-doc-mcp-server.log', 'level': 'DEBUG', 'rotation': '10 MB'},
```
Comment (scottschreckengaust) [inline src/code-doc-generation-mcp-server/awslabs/code_doc_generation_mcp_server/server.py]: same name as server?

File: src/code-doc-generation-mcp-server/awslabs/code_doc_generation_mcp_server/utils/templates.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+from loguru import logger
+
+
+# Mapping of filenames to template types
+TEMPLATE_FILE_MAPPING = {
+    'README.md': 'README',
+    'API.md': 'API',
+    'BACKEND.md': 'BACKEND',
+    'FRONTEND.md': 'FRONTEND',
+    'DEPLOYMENT_GUIDE.md': 'DEPLOYMENT',
```
Comment (scottschreckengaust) [inline src/code-doc-generation-mcp-server/awslabs/code_doc_generation_mcp_server/utils/templates.py]: consider a future `CONTRIBUTING.md` (not this PR)

File: src/code-doc-generation-mcp-server/awslabs/code_doc_generation_mcp_server/server.py (hunk 1 above)
Comment (jimini55) [reply src/code-doc-generation-mcp-server/awslabs/code_doc_generation_mcp_server/server.py]: Will remove the logger file configuration

**Review by awslabs-mcp — APPROVED** [review]
## Pull Request Review Summary

I've completed a thorough review of this pull request adding the new Code Documentation Generation MCP Server. The PR introduces a valuable addition to the MCP server ecosystem that helps users automatically generate comprehensive documentation for their code repositories.

### Strengths

1. **Well-structured implementation**: The code is organized following established MCP server patterns with clear separation of concerns between server.py, models.py, and utility modules.

2. **Comprehensive documentation**: The PR includes detailed README documentation that explains the server's architecture, features, and usage.

3. **Good test coverage**: The implementation includes unit tests for all main functionality.

4. **Integration with other MCP servers**: The server has well-designed integration points with the AWS Diagram MCP Server for enhanced documentation with diagrams.

5. **Workflow-based approach**: The server implements a clear step-by-step workflow (prepare → create_context → plan → generate) that guides users through the documentation process.

6. **Security considerations**: Uses defusedxml for secure XML parsing and proper error handling throughout the codebase.

7. **Proper adherence to DESIGN_GUIDELINES.md**: The implementation follows the project's design guidelines, including module structure, code organization, and documentation standards.

### Comments

The PR doesn't attempt to automatically write the documentation content but rather creates the structure and guides the AI assistant through filling it with appropriate content based on project analysis. This is a thoughtful design choice that leverages the strengths of both the MCP server (structure analysis) and the AI (content generation).

All required checks have passed, and the code appears ready to be merged.

### Conclusion

This PR adds a valuable tool for documentation generation that follows project standards and guidelines. I recommend approving and merging this contribution.

Run ID: 15003183906

**Review by awslabs-mcp — COMMENTED** [review]
## PR Review Assessment

I've reviewed PR #296 which adds the new Code Documentation Generation MCP Server. This server provides AI assistants with the capability to automatically analyze repository structures and generate comprehensive documentation.

### General Assessment
- The implementation is well-structured and follows the AWS Labs MCP server design guidelines
- The code quality is good with proper error handling and comprehensive tests
- The server successfully integrates with the repomix tool to extract repository structure
- Documentation is complete with clear examples and usage instructions

### Observations
1. **Package/Directory Naming Inconsistency**: There are inconsistencies between naming in the PR, where both "code-doc-gen-mcp-server" and "code-doc-generation-mcp-server" are being used. The current implementation in the repository is using "code-doc-gen-mcp-server", which appears to be the correct name based on the pyproject.toml.

2. **Code Coverage**: The Codecov report shows some missing coverage in utility classes, particularly in repomix_manager.py (73.46% coverage) and doc_generator.py (81.09% coverage). While this isn't a blocker, it would be good to add additional tests to improve coverage.

3. **Integration with Other MCP Servers**: The code does a good job of recommending integration with AWS Diagram MCP Server for architecture diagrams, which enhances the user experience.

4. **Documentation Templates**: The templates are well-structured with appropriate sections and instructions for different documentation types.

### Recommendations
This PR implements a valuable addition to the MCP server collection. I recommend resolving the naming inconsistencies between "code-doc-gen-mcp-server" and "code-doc-generation-mcp-server" before merging, as this could cause confusion for users and potential issues with documentation links. The repository structure and all references should use a consistent name.

Run ID: 15004351283

**Review by awslabs-mcp — CHANGES_REQUESTED** [review]
## Final Review Assessment

Thank you for submitting this valuable addition to the AWS Labs MCP server collection. The Code Documentation Generation MCP Server will be a useful tool for automatically generating comprehensive documentation for code repositories.

### Required Changes

1. **Package/Directory Naming Inconsistency**: The PR contains inconsistent naming - both \code-doc-gen-mcp-server\ and \code-doc-generation-mcp-server\ are used in different files. The correct name appears to be \code-doc-gen-mcp-server\ based on the pyproject.toml, but there are several files and references using the incorrect \code-doc-generation-mcp-server\ naming. Please update all files and references to use a consistent name before merging.

2. **Code Coverage**: Consider adding additional tests to improve coverage for the utility classes, particularly in repomix_manager.py and doc_generator.py, which currently have 73.46% and 81.09% coverage respectively.

Once these issues are addressed, the PR can be approved. The implementation design and overall code quality meet the expected standards for AWS Labs MCP servers.

Run ID: 15004351283

**Comment by jimini55** [comment]
Renamed code-doc-generation-mcp-server to code-doc-gen-mcp-server to keep the name short. I kept `Code Documentation Generation MCP Server` naming in READMEs and other docs, but in code/directory code-doc-gen. Tested with the new name.

**Comment by jimini55** [comment]
> ## Final Review Assessment
> Thank you for submitting this valuable addition to the AWS Labs MCP server collection. The Code Documentation Generation MCP Server will be a useful tool for automatically generating comprehensive documentation for code repositories.
> 
> ### Required Changes
> 1. **Package/Directory Naming Inconsistency**: The PR contains inconsistent naming - both \code-doc-gen-mcp-server\ and \code-doc-generation-mcp-server\ are used in different files. The correct name appears to be \code-doc-gen-mcp-server\ based on the pyproject.toml, but there are several files and references using the incorrect \code-doc-generation-mcp-server\ naming. Please update all files and references to use a consistent name before merging.
> 2. **Code Coverage**: Consider adding additional tests to improve coverage for the utility classes, particularly in repomix_manager.py and doc_generator.py, which currently have 73.46% and 81.09% coverage respectively.
> 
> Once these issues are addressed, the PR can be approved. The implementation design and overall code quality meet the expected standards for AWS Labs MCP servers.
> 
> Run ID: 15004351283

Renamed `code-doc-generation-mcp-server` to `code-doc-gen-mcp-server` to keep the name short. I kept `Code Documentation Generation MCP Server` naming in READMEs and other docs, but in code/directory code-doc-gen. Tested with the new name.

**Review by awslabs-mcp — APPROVED** [review]
## Pull Request Review Summary

I've completed a thorough review of the Code Documentation Generation MCP Server PR. This new MCP server provides automated repository structure analysis and documentation generation that will be a valuable addition to the AWS MCP Server collection.

### Code Quality and Implementation

The implementation follows AWS Labs MCP server design guidelines with clear separation of concerns between:
- `server.py` - Main MCP tool definitions and server setup
- `models.py` - Well-structured Pydantic models
- Utility modules for repo analysis and document generation

The code is well-organized with a logical four-step workflow:
1. `prepare_repository` - Analyzes directory structure
2. `create_context` - Sets up documentation context
3. `plan_documentation` - Creates documentation plan
4. `generate_documentation` - Generates document structures

The implementation includes proper error handling, comprehensive tests, and good documentation practices.

### Documentation

The README is comprehensive and clearly explains:
- Server architecture and workflow
- Installation and usage instructions
- Integration with other MCP servers
- Data models and tool interfaces

### Testing

Test coverage is good with unit tests for all main components:
- Server functions and MCP tools
- RepomixManager functionality
- Document generation utilities
- Edge cases and error handling

### Integration

The server properly integrates with the repository by:
- Following established naming conventions
- Using appropriate dependencies
- Including proper licensing headers
- Documenting integration with other MCP servers (particularly AWS Diagram MCP Server)

### Package Naming Consistency

One issue previously identified was naming inconsistency between "code-doc-gen-mcp-server" and "code-doc-generation-mcp-server". I'm pleased to see this has been addressed, and the PR now consistently uses "code-doc-gen-mcp-server" throughout all files, including:
- Directory name
- Package name in pyproject.toml
- Import statements
- Documentation links

All required checks have passed, and the PR is ready for merging.

Run ID: 15006737457

**Review by scottschreckengaust — APPROVED** [review]
LGTM
